Python/import.c

```diff
@@ -919,167 +919,134 @@ extensions_lock_release(void)
 static void
 _extensions_cache_init(void)
 {
-    /* The runtime (i.e. main interpreter) must be initializing,
-       so we don't need to worry about the lock. */
-    _PyThreadState_InitDetached(&EXTENSIONS.main_tstate,
-                                _PyInterpreterState_Main());
+    /* The lock is initialized directly with the general runtime state. */
+    assert(EXTENSIONS.mutex != NULL);
+    //assert(EXTENSIONS.head == NULL);
```

Comment:
Sort of.  I had expected the assert to work (and it would in a simpler world).  I was going to circle back to it but, having slept on it, don't see the need.  It can be dropped.